cli/pkg/kctrl/cmd/package/release/helm_openapi_schema_gen_test.go

```diff
@@ -0,0 +1,221 @@
+    description: Float example
+    format: float
+    type: number
+  image:
+    default: my-docker-image:1.0.0
+    description: Image details
+    type: string
+  intExample:
+    default: 3
+    description: Integer example
+    type: integer
+  name:
+    default: test-container
+    description: Container name
+    type: string
+type: object
+`},
+		{
+			name: "nested complex object",
+			input: `
+containers:
+  # Image details
+  image: my-docker-image:1.0.0
+  # Array example
+  env:
+  # key1 example
+  - key1: val1
+  -
+  # key2 example
+    key2: val2
+  # Float example
+  floatExample: 2.3
+`,
+			want: `properties:
+  containers:
+    properties:
+      env:
+        default: []
+        description: Array example
+        items:
+          properties:
+            key1:
+              default: val1
+              description: example
+              type: string
+            key2:
+              default: val2
+              description: example
+              type: string
+          type: object
+        type: array
+      floatExample:
+        default: 2.3
+        description: Float example
+        format: float
+        type: number
+      image:
+        default: my-docker-image:1.0.0
+        description: Image details
+        type: string
+    type: object
+type: object
+`},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			defer cleanup()
+			dirName := "tmp"
+			err := createDir(dirName)
+			require.NoError(t, err)
+			fileName := "values.yaml"
+			err = createFile(filepath.Join(dirName, fileName), []byte(test.input))
+			require.NoError(t, err)
+			valuesSchema, err := release.NewHelmValuesSchemaGen("tmp").Schema()
+			output, err := yaml.JSONToYAML(valuesSchema.OpenAPIv3.Raw)
+			require.Equal(t, test.want, string(output), "Expected valuesSchema to match")
+		})
+	}
+}
+
+func TestHelmValuesSchemaGen_Schema_EmptyFile(t *testing.T) {
+	defer cleanup()
+	dirName := "tmp"
+	err := createDir(dirName)
+	require.NoError(t, err)
+	fileName := "values.yaml"
+	err = createFile(filepath.Join(dirName, fileName), []byte(""))
+	require.NoError(t, err)
+	valuesSchema, err := release.NewHelmValuesSchemaGen("tmp").Schema()
+	require.Equal(t, 0, len(valuesSchema.OpenAPIv3.Raw), "Expected valuesSchema.OpenAPIv3.Raw to be empty")
+}
+
+func TestHelmValuesSchemaGen_Schema_File_Not_Present(t *testing.T) {
+	defer cleanup()
+	cleanup()
+	dirName := "tmp"
+	err := createDir(dirName)
+	require.NoError(t, err)
+	valuesSchema, err := release.NewHelmValuesSchemaGen("tmp").Schema()
+	require.Equal(t, 0, len(valuesSchema.OpenAPIv3.Raw), "Expected valuesSchema.OpenAPIv3.Raw to be empty")
+}
+
+func createDir(dirName string) error {
+	err := os.Mkdir(dirName, fs.ModePerm)
+	if err != nil {
+		return fmt.Errorf("unable to create %s directory: %s", dirName, err.Error())
+	}
+	return nil
+}
+
+func createFile(filePath string, fileData []byte) error {
+	err := os.WriteFile(filePath, fileData, fs.ModePerm)
```

Comment:
I think we can directly use this function (Same for `Mkdir`).

Reply:
Resolved